### PR TITLE
feat: expose dotenv API

### PR DESCRIPTION
**Pull request by antongolub**
continues #974

```js
const multiline = `SIMPLE=xyz123
NON_INTERPOLATED='raw text without variable interpolation'
MULTILINE = """
long text here,
e.g. a private SSH key
"""
ENV=v1\nENV2=v2\n\n\n  ENV3  =    v3   \n   export ENV4=v4
`

assert.deepEqual(parseDotenv(multiline), {
  SIMPLE: 'xyz123',
  NON_INTERPOLATED: 'raw text without variable interpolation',
  MULTILINE: '\nlong text here,\ne.g. a private SSH key\n',
  ENV: 'v1',
  ENV2: 'v2',
  ENV3: 'v3',
  ENV4: 'v4',
})
```

fyi, @1e9y , @easymikey 

- [x] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -358,15 +358,58 @@ export const toCamelCase = (str: string) =>
...
+  const kr = /^[a-zA-Z_]+[a-zA-Z0-9_]*$/
+  const sr = /\s/
+  let k = ''
+  let b = ''
+  let q = ''
+  let i = 0
+  const cap = () => { if (b && k) {
+    if (!kr.test(k)) throw new Error(`Invalid identifier: ${k}`)
+    e[k] = b; b = ''; k = ''
+  }}
```
Comment (easymikey) [inline src/util.ts:373]: Maybe like this?

```suggestion
const cap = () => {
  if (b && k) {
    if (!kr.test(k)) throw new Error(`Invalid identifier: ${k}`);
    e[k] = b; b = k = "";
  }
};
```

File: src/util.ts [hunk 2]
```diff
@@ -358,15 +358,58 @@ export const toCamelCase = (str: string) =>
...
-    return r
-  }, {})
+// prettier-ignore
+export const parseDotenv = (content: string): NodeJS.ProcessEnv => {
+  const kr = /^[a-zA-Z_]+\w*$/
+  const sr = /\s/
+  const e: Record<string, string> = {}
+  let k = ''
+  let b = ''
+  let q = ''
```
Comment (1e9y) [inline src/util.ts:368]: Why not use more readable variable names?

Comment (antongolub) [reply src/util.ts:368]: Just a minification challenge without any special reason.

(s)pace(r)egex
(k)ey(r)egex

(e)nv
(k)ey
(b)uffer
(q)oute
(i)gnore
(c)har

Comment (1e9y) [reply src/util.ts:368]: Lets make it contributor-friendly and leave minification for Vite plugin!

**Comment by antongolub** [comment]
dotenv requires ~10Kb extra

image

**Comment by antonmedv** [comment]
I’m against adding support for this feature.

**Comment by antongolub** [comment]
I'm afraid, dotenv format requires:
* https://github.com/motdotla/dotenv/blob/master/tests/.env-multiline
* https://hexdocs.pm/dotenvy/dotenv-file-format.html

**Pull request merged**